File: CS/Serialization/EventArgs.cs
using System;

namespace Serialization
{
	public class AllowSerializePropertyEventArgs : EventArgs
	{
		private object serializationObject;
		private string propertyName;
		private Type propertyType;
		private object propertyValue;
		private bool serializeProperty;

		public AllowSerializePropertyEventArgs(object serializationObject, string propertyName, Type propertyType, object propertyValue)
		{
			this.serializationObject = serializationObject;
			this.propertyName = propertyName;
			this.propertyType = propertyType;
			this.propertyValue = propertyValue;
			this.SerializeProperty = true;
		}

		public object SerializationObject
		{
			get
			{
				return serializationObject;
			}
		}

		public string PropertyName
		{
			get
			{
				return propertyName;
			}
		}

		public Type PropertyType
		{
			get
			{
				return propertyType;
			}
		}

		public object PropertyValue
		{
			get
			{
				return propertyValue;
			}
		}

		public bool SerializeProperty
		{
			get
			{
				return serializeProperty;
			}
			set
			{
				serializeProperty = value;
			}
		}
	}

	public class SerializeCustomPropertiesEventArgs : EventArgs
	{
		private object serializationObject;
		private string[] propertyNames;

		public SerializeCustomPropertiesEventArgs(object serializationObject)
		{
			this.serializationObject = serializationObject;
		}

		public object SerializationObject
		{
			get
			{
				return serializationObject;
			}
		}

		public string[] PropertyNames
		{
			get
			{
				return propertyNames;
			}
			set
			{
				propertyNames = value;
			}
		}
	}

	public class CustomPropertyValueEventArgs : EventArgs
	{
		private object serializationObject;
		private string propertyName;
		private object propertyValue = null;

		public CustomPropertyValueEventArgs(object serializationObject, string propertyName)
		{
			this.serializationObject = serializationObject;
			this.propertyName = propertyName;
		}

		public object SerializationObject
		{
			get
			{
				return serializationObject;
			}
		}

		public string PropertyName
		{
			get
			{
				return propertyName;
			}
		}

		public object PropertyValue
		{
			get
			{
				return propertyValue;
			}
			set
			{
				propertyValue = value;
			}
		}
	}
}

File: VB/Serialization/EventArgs.vb
Imports Microsoft.VisualBasic
Imports System

Namespace Serialization
	Public Class AllowSerializePropertyEventArgs
		Inherits EventArgs
		Private serializationObject_Renamed As Object
		Private propertyName_Renamed As String
		Private propertyType_Renamed As Type
		Private propertyValue_Renamed As Object
		Private serializeProperty_Renamed As Boolean

		Public Sub New(ByVal serializationObject As Object, ByVal propertyName As String, ByVal propertyType As Type, ByVal propertyValue As Object)
			Me.serializationObject_Renamed = serializationObject
			Me.propertyName_Renamed = propertyName
			Me.propertyType_Renamed = propertyType
			Me.propertyValue_Renamed = propertyValue
			Me.SerializeProperty = True
		End Sub

		Public ReadOnly Property SerializationObject() As Object
			Get
				Return serializationObject_Renamed
			End Get
		End Property

		Public ReadOnly Property PropertyName() As String
			Get
				Return propertyName_Renamed
			End Get
		End Property

		Public ReadOnly Property PropertyType() As Type
			Get
				Return propertyType_Renamed
			End Get
		End Property

		Public ReadOnly Property PropertyValue() As Object
			Get
				Return propertyValue_Renamed
			End Get
		End Property

		Public Property SerializeProperty() As Boolean
			Get
				Return serializeProperty_Renamed
			End Get
			Set(ByVal value As Boolean)
				serializeProperty_Renamed = value
			End Set
		End Property
	End Class

	Public Class SerializeCustomPropertiesEventArgs
		Inherits EventArgs
		Private serializationObject_Renamed As Object
		Private propertyNames_Renamed() As String

		Public Sub New(ByVal serializationObject As Object)
			Me.serializationObject_Renamed = serializationObject
		End Sub

		Public ReadOnly Property SerializationObject() As Object
			Get
				Return serializationObject_Renamed
			End Get
		End Property

		Public Property PropertyNames() As String()
			Get
				Return propertyNames_Renamed
			End Get
			Set(ByVal value As String())
				propertyNames_Renamed = value
			End Set
		End Property
	End Class

	Public Class CustomPropertyValueEventArgs
		Inherits EventArgs
		Private serializationObject_Renamed As Object
		Private propertyName_Renamed As String
		Private propertyValue_Renamed As Object = Nothing

		Public Sub New(ByVal serializationObject As Object, ByVal propertyName As String)
			Me.serializationObject_Renamed = serializationObject
			Me.propertyName_Renamed = propertyName
		End Sub

		Public ReadOnly Property SerializationObject() As Object
			Get
				Return serializationObject_Renamed
			End Get
		End Property

		Public ReadOnly Property PropertyName() As String
			Get
				Return propertyName_Renamed
			End Get
		End Property

		Public Property PropertyValue() As Object
			Get
				Return propertyValue_Renamed
			End Get
			Set(ByVal value As Object)
				propertyValue_Renamed = value
			End Set
		End Property
	End Class
End Namespace
